avro2tf/src/main/scala/com/linkedin/avro2tf/parsers/TensorizeInJobParamsParser.scala

```diff
@@ -266,21 +285,45 @@ object TensorizeInJobParamsParser {
         enableCache = false,
         skipConversion = false,
         outputFormat = AVRO_RECORD,
-        extraColumnsToKeep = Seq.empty
+        extraColumnsToKeep = Seq.empty,
+        tensorsSharingFeatureLists = None
       )
     ) match {
       case Some(params) =>
         // Check if users only specify either date range or days range
         if (params.inputDateRange.nonEmpty && params.inputDaysRange.nonEmpty) {
           throw new IllegalArgumentException("Please only specify either date range or days range.")
         }
-        if(!params.isTrainMode && params.executionMode == TrainingMode.training){
+        params.tensorsSharingFeatureLists match {
```

Comment:
should move this logic (line 297-315) inside the parser, since it does not depend on the values of other params

Reply:
Good point. Will address.